src/Build/BackEnd/Components/RequestBuilder/TargetUpToDateChecker.cs

```diff
@@ -960,6 +963,13 @@ private static List<string> GetItemSpecsFromItemVectors(ItemVectorPartitionColle
         /// <returns>true, if any "input" is newer than any "output", or if any input or output does not exist.</returns>
         internal static bool IsAnyOutOfDate<T>(out DependencyAnalysisLogDetail dependencyAnalysisDetailEntry, string projectDirectory, IList<T> inputs, IList<T> outputs)
         {
+#if FEATURE_ENUMERATION
```

Comment:
You could make this a Trait to make it easier to turn on via an environment variable